Update auth documentation to document kubernetes.io/* secrets (#1054)
tektoncd/pipeline: docs/auth.md

credentials are then used to authenticate when retrieving any
`PipelineResources`.

## Kubernetes's Docker registry's secret

Kubernetes defines two types of secrets for Docker registries :

> `kubernetes.io/dockercfg` and `kubernetes.io/dockerconfigjson`. Tekton supports

Reviewer: ```suggestion
the old format `kubernetes.io/dockercfg` and the new format `kubernetes.io/dockerconfigjson`. Tekton supports
```

those secrets in addition to the one described above.

1. Define a `Secret` from a Docker client configuration file, as documented in
   [Pull an Image from a Private Registry](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/)

   ```bash
   kubectl create secret generic regcred \
    --from-file=.dockerconfigjson=<path/to/.docker/config.json> \
    --type=kubernetes.io/dockerconfigjson
   ```

> 1. Direct a `ServiceAccount` to use this `Secret`:

Reviewer: ```suggestion
1. Instruct a `ServiceAccount` to use this `Secret`:
```

   ```yaml
   apiVersion: v1
   kind: ServiceAccount
   metadata:
     name: build-bot
   secrets:
     - name: regcred
   ```

> 1. Use that `ServiceAccount` in your `Build`:

Reviewer 2: Should this be a `Task`?

Author: :man_facepalming: @dibyom you are right ! :sweat_smile: :joy:

   ```yaml
   apiVersion: tetkon.dev/v1alpha1

>    kind: Build

Reviewer 2: Same as above

   metadata:
     name: build-with-basic-auth
   spec:
     serviceAccountName: build-bot
     steps:
     ...
   ```

1. Execute the build:

   ```shell

>    kubectl apply --filename secret.yaml --filename serviceaccount.yaml --filename build.yaml

Reviewer: ```suggestion
   kubectl apply --filename secret.yaml --filename serviceaccount.yaml --filename task.yaml
```

   ```

> When this build executes, before steps execute, a `~/.docker/config.json` will

Reviewer: ```suggestion
When this build executes, before the steps are getting executed, a `~/.docker/config.json` will
```
